growattServer/__init__.py

```diff
@@ -24,12 +25,45 @@ class Timespan(IntEnum):
 
 class GrowattApi:
     server_url = 'https://server-api.growatt.com/'
+    user_agent=''
 
-    def __init__(self):
+    def __init__(self, generate_user_agent=False, user_agent='Dalvik/2.1.0 (Linux; U; Android 12; Device)'):
         self.session = requests.Session()
-        headers = {'User-Agent': 'Dalvik/2.1.0 (Linux; U; Android Device)'}
+        self.user_agent = user_agent
+
+        if generate_user_agent:
+          self.__generate_user_agent()
+
+        headers = {'User-Agent': self.user_agent}
         self.session.headers.update(headers)
 
+    def __generate_user_agent(self):
+      #Start by setting the generated version to our default in case we can't contact the server to generate a random user-agent
+      generated_user_agent = self.user_agent
+
+      #Request a list of Dalvik user-agents from a random android version
+      android_versions = [9,10,11,12,13]
+      and_vers = android_versions[random.randint(0,len(android_versions)-1)]
+      user_agents_url='https://user-agents.net/download'
+
+      post_data={
+        'browser': 'dalvik',
+        'name': '2.1.0 (Linux; U; Android ' + str(and_vers),
+        'download': 'json'
+      }
+
+      user_agent_response = requests.post(user_agents_url, data=post_data)
```

Comment:
This does give the most authentic results for user agents, but i'm not a huge fan of relying on an external service for it.
Not solely for possible outages, which would probably be caught by your try catch. But also slow responses from the other end. If the network or server is struggling a little initialising this class might take several seconds

I don't expect them to actually have checks in place to verify the validity of the user agent considering the formats they can take and they can take many forms e.g. bingbot
`Mozilla/5.0 (compatible; bingbot/2.0; +http://www.bing.com/bingbot.htm)`
We could add the url to the repo, and maybe randomly generate a browser version

Reply:
I completely understand and I'm also reluctant to make the change as well, I just thought it would be best to implement something that looked as 'realistic' as possible.
Remember that I've made this so that it's completely 'optional' for people to use, the default is still to use the initialised value. You have to set the auto-generation to `True` to actually use that method.
The alternative is that we strip out the autogeneration and people can do that externally and then pass it in as an optional initialisation value for the library.
We could even provide an example on how to do that if people wanted to.
What do you think about that option?

Reply:
Maybe instead of trying to generate a 'realistic' user agent in here we should allow people to choose to introduce randomness into the one that is provided in the initialisation of the function. So it would just generate a random 10 character string to put on the end when initialised, that way when/if growatt block the string again all they have to do is restart the library and it'll generate a new one.

Reply:
Yeah that's what i'm thinking may be the best bet right now, to add a random string in the user agent like
`2.1.0 (Linux; U; Android 11; grotwattserver/1234; +https://github.com/indykoning/PyPi_GrowattServer)`
where the 2134 is our random string

Reply:
ok, I've got some time tomorrow evening, so I'll throw another option together and submit it as a separate PR so we can compare them both and see what you think.

Reply:
I've pushed an alternative option, let me know what you think: https://github.com/indykoning/PyPi_GrowattServer/pull/42